frontend/src/metabase/dashboard/components/DashCard/DashCardMenu/DashCardMenu.tsx

```diff
@@ -60,72 +47,80 @@ const DashCardMenu = ({
   uuid,
   token,
   onEditQuestion,
-  onDownloadResults,
 }: DashCardMenuProps) => {
   const store = useStore();
 
-  const [{ loading }, handleDownload] = useAsyncFn(
-    async (opts: { type: string; enableFormatting: boolean }) => {
-      const params = getParameterValuesBySlugMap(store.getState());
-
-      await onDownloadResults({
-        ...opts,
-        question,
-        result,
-        dashboardId,
-        dashcardId,
-        uuid,
-        token,
-        params,
-      });
+  const [{ loading }, handleDownload] = useDownloadData({
```

Comment:
We duplicate this `handleDownload` function in `QueryPopoverWidget` so I moved it to a hook